packages/inspector/src/components/EntityInspector/ScriptInspector/ScriptInspector.tsx

```diff
@@ -0,0 +1,427 @@
+    },
+    [scripts, setComponentValue],
+  );
+
+  // memoize parsed layouts to avoid re-parsing on every render
+  const parsedLayouts = useMemo(() => {
+    return scripts.map(script => parseLayout(script.layout));
+  }, [scripts]);
+
+  const createScript = useCallback(
+    (path: string, priority = 0, content: string) => {
+      const { params, error } = getScriptParams(content);
+      const layout: ScriptLayout = { params, error };
+
+      const newScript: ScriptItem = {
+        path,
+        priority,
+        layout: JSON.stringify(layout),
+      };
+
+      addScript(newScript);
+      setEmptyScriptModuleMode(undefined);
+    },
+    [addScript, setEmptyScriptModuleMode],
+  );
+
+  const handleRemove = useCallback(async () => {
+    sdk.operations.removeComponent(entityId, Script);
+    await sdk.operations.dispatch();
+  }, [sdk, entityId, Script]);
+
+  const handleRemoveScript = useCallback(
+    (index: number) => {
+      removeScript(index);
+    },
+    [removeScript],
+  );
+
+  const handleEditScript = useCallback(
+    async (index: number) => {
+      try {
+        const script = scripts[index];
+        if (!script) return;
+
+        const sceneClient = getSceneClient();
+        if (!sceneClient) return;
+
+        await sceneClient.openFile(script.path);
+      } catch (error) {
+        console.error('Failed to open script:', error);
+      }
+    },
+    [scripts],
+  );
+
+  // this will reload ALL scripts, not only the current entity ones...
+  const handleReloadScripts = useCallback(
+    async (e: React.MouseEvent<SVGElement>) => {
+      e.stopPropagation();
+      setError(undefined);
+
+      const dataLayer = getDataLayerInterface();
+      if (!dataLayer) return;
+
+      let firstError: string | undefined;
+      const allEntitiesWithScripts = Array.from(sdk.engine.getEntitiesWith(Script));
+
+      if (allEntitiesWithScripts.length === 0) return;
+
+      await Promise.all(
+        allEntitiesWithScripts.map(async ([entity, scriptComponent]) => {
+          const entityScripts = scriptComponent.value || [];
+          if (entityScripts.length === 0) return;
+
+          const updatedScripts = await Promise.all(
+            entityScripts.map(async script => {
+              try {
+                const content = await readScript(dataLayer, script.path);
+                const newLayout = getScriptParams(content);
+                const currentLayout = parseLayout(script.layout) || { params: {} };
+                const layout = mergeLayout(newLayout, currentLayout);
+
+                return {
+                  ...script,
+                  layout: JSON.stringify(layout),
+                };
+              } catch (error) {
+                const msg = `Failed to reload script '${script.path}'`;
+                console.error(`${msg}:`, error);
+                if (!firstError) firstError = msg;
+                return script; // keep existing if read fails
```

Comment:
could the associated file script be deleted in the filesystem, and we still preserve it?

<img width="290" height="404" alt="Image" src="https://github.com/user-attachments/assets/fa8cad12-e467-436d-b2b9-768417dbfe8b" />